src/main/java/com/uber/cadence/testing/WorkflowShadower.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Modifications Copyright (c) 2017-2021 Uber Technologies Inc.
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+package com.uber.cadence.testing;
+
+import com.google.common.collect.Lists;
+import com.uber.cadence.WorkflowExecution;
+import com.uber.cadence.internal.shadowing.ReplayWorkflowActivity;
+import com.uber.cadence.internal.shadowing.ReplayWorkflowActivityImpl;
+import com.uber.cadence.internal.shadowing.ScanWorkflowActivity;
+import com.uber.cadence.internal.shadowing.ScanWorkflowActivityImpl;
+import com.uber.cadence.serviceclient.IWorkflowService;
+import com.uber.cadence.shadower.Mode;
+import com.uber.cadence.shadower.ReplayWorkflowActivityParams;
+import com.uber.cadence.shadower.ReplayWorkflowActivityResult;
+import com.uber.cadence.shadower.ScanWorkflowActivityParams;
+import com.uber.cadence.shadower.ScanWorkflowActivityResult;
+import com.uber.cadence.worker.ShadowingOptions;
+import com.uber.m3.tally.NoopScope;
+import com.uber.m3.tally.Scope;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+
+public final class WorkflowShadower {
+
+  private final ShadowingOptions options;
+  private final String query;
+  private final ScanWorkflowActivity scanWorkflow;
+  private final ReplayWorkflowActivity replayWorkflow;
+
+  public WorkflowShadower(IWorkflowService service, ShadowingOptions options, String taskList) {
```

Comment:
I think we don't support shadow options with concurrency > 1 for local shadower. Is there a check for it and will an exception be raised?

Reply:
add a check